src/main/java/onlinebookstore/service/impl/BookServiceImpl.java

```diff
@@ -11,14 +14,24 @@
 @RequiredArgsConstructor
 public class BookServiceImpl implements BookService {
     private final BookRepository bookRepository;
+    private final BookMapper bookMapper;
 
     @Override
-    public Book save(Book book) {
-        return bookRepository.save(book);
+    public BookDto save(CreateBookRequestDto bookRequestDto) {
+        Book savedBook = bookRepository
+                .save(bookMapper.toModel(bookRequestDto));
+        return bookMapper.toDto(savedBook);
     }
 
     @Override
-    public List<Book> findAll() {
-        return bookRepository.findAll();
+    public List<BookDto> findAll() {
+        return bookRepository.findAll().stream()
+                .map(bookMapper::toDto)
+                .toList();
+    }
+
+    @Override
+    public BookDto findById(Long id) {
+        return bookMapper.toDto(bookRepository.findById(id));
```

Comment:
when you'll update the repository to return `Optional` - don't use `ifPresent` here

there is a `map` method, so that you can do everything in a single chain